include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -186,7 +186,17 @@ struct disjoint_segment_box_sphere_or_spheroid
         bool b2 = formula::azimuth_side_value(alp1, a_b2) > 0;
         bool b3 = formula::azimuth_side_value(alp1, a_b3) > 0;
 
-        if (!(b0 && b1 && b2 && b3) && (b0 || b1 || b2 || b3))
+        bool all_ones = b0 && b1 && b2 && b3;
+        bool all_zeros = !(b0 || b1 || b2 || b3);
```

Comment:
`azimuth_side_value` can return values in {-1, 0, 1} meaning {right, on_segment, left} respectively so this condition doesn't mean `all_zeros` It means `not_all_ones` or `either_minus_ones_or_zeros`.

Reply:
By "zeros" and "ones" I had in mind the values of b1, ..., b3. I see how this is misleading. I rephrased it.